server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -109,6 +109,16 @@ public TermsAggregationBuilder(String name, ValueType valueType) {
         super(name, ValuesSourceType.ANY, valueType);
     }
 
+    protected TermsAggregationBuilder(TermsAggregationBuilder clone) {
+        super(clone);
+        this.order = clone.order;
+        this.executionHint = clone.executionHint;
+        this.includeExclude = clone.includeExclude;
+        this.collectMode = clone.collectMode;
+        this.bucketCountThresholds = clone.bucketCountThresholds;
```

Comment:
It seems that BucketCountThresholds is mutable so I think we need to clone it as well?